dbcon/execplan/calpontsystemcatalog.cpp

```diff
@@ -6319,5 +6328,47 @@ bool ctListSort(const CalpontSystemCatalog::ColType& a, const CalpontSystemCatal
   return a.colPosition < b.colPosition;
 }
 
+bool operator <(const Partitions& a, const Partitions& b)
+{
+  // lexicographic order.
+  uint32_t l = std::min(a.fPartNames.size(), b.fPartNames.size());
+  for (uint32_t i = 0; i < l; i++)
+  {
+    if (a.fPartNames[i] < b.fPartNames[i])
+    {
+      return true;
+    }
+    if (a.fPartNames[i] > b.fPartNames[i])
+    {
+      return false;
+    }
+  }
+  if (l > a.fPartNames.size())
```

Comment:
How this and line 6350 can be true? `l` is always >= to the size of fPartNames IMHO. Am I missing something here?

Reply:
Yes, it should be `i`.

Reply:
No, `i` is equal to `l` here, the comparisons should be opposite - not "is `l` greater than `some.size()`," but "is `l` less than `some.size()`".

Reply:
Fixed.